services/DynamoService.test.ts

```diff
@@ -85,36 +82,45 @@ describe("DynamoService", () => {
   });
 
   describe('putItem', () => {
-    const item = { commonName: 'Treebeard', sidarinName: 'Fangorn', entishName: 'Not stored due to buffer overflow', orcishName: '' };
+    // We are marshalling the item as that is what we expect to receive from the User model
+    const item = DynamoDB.Converter.marshall(
+      {
+        commonName: 'Treebeard',
+        entishName: 'Not stored due to buffer overflow',
+        orcishName: '',
+        sidarinName: 'Fangorn',
+      },
+      { convertEmptyValues: true },
+    );
     const tableName = 'Ents';
 
     it('puts to a DynamoDB table', async () => {
       await service.putItem(item, tableName);
-      expect(mockPut).toHaveBeenCalledWith({ Item: item, TableName: tableName }, expect.any(Function));
+      expect(mockPut).toHaveBeenCalledWith(
+        { Item: item, TableName: tableName },
+        expect.any(Function),
+      );
     });
 
     // The DynamoDB API breaks if empty strings are passed in
     it('converts empty strings in user model to nulls', async () => {
       await service.putItem(item, tableName);
-      expect(mockPut.mock.calls[0][0]['Item']['orcishName']).toEqual(null);
+      expect(mockPut.mock.calls[0][0].Item.orcishName).toEqual({ NULL: true });
```

Comment:
huh?

Reply:
TypeScript did not like the way User#saveToDynamo was passing in the dynamoItem to `service.putItem` so I had to update the way it worked to be more in line with the AWS recommended usage, which is marshalling the item before saving it to the dynamoDB. `null` items become marshalled to `{ NULL: true }`. Sorry that wasn't explained in the PR.